src/btree/bt_sync_obsolete.c

```diff
@@ -391,6 +407,24 @@ __checkpoint_cleanup_page_skip(
         return (0);
     }
 
+    if (__sync_obsolete_limit_reached(session)) {
```

Comment:
This change now limits the checkpoint cleanup operating the obsolete pages. Whenever we reach the obsolete time window pages, we start to limit the obsolete page cleanup. 

How about doing all these checks after the obsolete page checks?

Reply:
As discussed, will move the condition lower.

Reply:
Please see the refactoring 1316dc1512fe1126d395d0c497d9aaaea9502d8e